index.js

```diff
@@ -116,6 +116,10 @@ module.exports = {
     // regular expressions.
     'unicorn/no-unsafe-regex': 'error',
 
+    // This is a common pattern in React, and has been proven to be a premature optimization
```

Comment:
can you provide a link for the "proven" portion? the rule itself has two links to support it's claims on perf, it'd be good to provide a counterargument if/when this rule comes into discussion in the future 🙏

Reply:
maybe a warning would be sufficient? probably depends on the analysis?

Reply:
I can't seem to find much supporting evidence either way